recipes/azure-storage-cpplite/meta.yaml

```diff
@@ -0,0 +1,75 @@
+{% set name = "azure-storage-cpplite" %}
+{% set version = "0.3.0" %}
+
+package:
+  name: {{ name|lower }}-split
+  version: {{ version }}
+
+source:
+  url: https://github.com/Azure/{{ name }}/archive/v{{ version }}.tar.gz
+  sha256: 25f34354fb0400ffe1b5a5c09c793c9fc8104d375910f6c84ab10fa50c0059cb
+
+build:
+  number: 0
+
+outputs:
+  - name: azure-storage-cpplite
+    script: build-shared.sh  # [unix]
+    script: bld-shared.bat  # [win]
+    build:
+      activate_in_script: True
+      run_exports:
+        - {{ pin_subpackage("azure-storage-cpplite", max_pin="x.x.x") }}
+    requirements:
+      build:
+        - {{ compiler('cxx') }}
+        - cmake
+        - make  # [unix]
+      host:
+        - libuuid  # [unix]
+        - libcurl
+        - openssl
+    test:
+      commands:
+        - test -f $PREFIX/include/blob/blob_client.h  # [unix]
+        - test -f $PREFIX/lib/libazure-storage-lite${SHLIB_EXT}  # [unix]
+        - test ! -f $PREFIX/lib/libazure-storage-lite.a  # [unix]
+        - if not exist %PREFIX%\\Library\\include\\blob\\blob_client.h exit 1  # [win]
+        - if not exist %PREFIX%\\Library\\bin\\azure-storage-lite.dll exit 1  # [win]
+        - if not exist %PREFIX%\\Library\\lib\\azure-storage-lite.lib exit 1  # [win]
+  - name: azure-storage-cpplite-static
+    script: build-static.sh  # [unix]
+    script: bld-static.bat  # [win]
+    build:
+      activate_in_script: True
+    requirements:
+      build:
+        - {{ compiler('cxx') }}
+        - cmake
+        - make  # [unix]
+      host:
+        - libuuid  # [unix]
+        - libcurl-static
+        - libopenssl-static
+        - {{ pin_subpackage("azure-storage-cpplite", exact=True) }}
+      run:
+        - {{ pin_subpackage("azure-storage-cpplite", exact=True) }}
+    test:
+      commands:
+        - test -f $PREFIX/include/blob/blob_client.h  # [unix]
+        - test -f $PREFIX/lib/libazure-storage-lite.a  # [unix]
+        - if not exist %PREFIX%\\Library\\include\\blob\\blob_client.h exit 1  # [win]
+        - if not exist %PREFIX%\\Library\\lib\\azure-storage-lite.lib exit 1  # [win]
```

Comment:
This is clobbering the shared import lib, not sure how to solve this.

Reply:
@wolfv @JohanMabille Do you have made any fixes in this regards when you were adding static libs to other feedstocks?

Reply:
From what I remember I didn't have to do anything, the static libraries that I've added to feedstock were already following a common pattern for static libraries on Windows: `name-RT-s-VERSION`, where RT is the runtim use to build the library (usually `MT`).

For this one I would submit a patch upstream. In the meantime, it is possible to rename the static library in the feedstock (but this requires patching cmake target and export files if they exist).

Reply:
I'm not sure I'm following what this issue is here.

Reply:
When you build a shared library for xxx on Windows, two files are generated: xxx.dll, which contains the code to execute, and xxx.lib which contains the exported symbols, use to link with.

The issue is that if you build a static library for xxx with the default options, you also generate a xxx.lib (not the same content as the previous one, but the same name and same location). If you install both shared and static packages, the latter.lib file overwirtes the former. Therefore, you need to use different names for static and share libraries.

Besides, applications linking with a static library should use the same runtime as that of the static library. It is therefore a common pattern to specify the runtime of the static library in its name.